src/server/api/controllers/profile.controller.js

```diff
@@ -37,7 +37,31 @@ const getAdminsProfile = async (userId) => {
     return error.message;
   }
 };
+
+const postAdminsProfile = async (adminRole, userId) => {
+  try {
+    const userProfileData = await knex.raw(
+      ` SELECT us.id AS userID, us.name as userName FROM users AS us, user_roles AS ur, roles AS rl WHERE ur.user_id = us.id AND ur.role_id = rl.id AND rl.name = 'student' AND us.id = ? `,
```

Comment:
Did you mean `rl.name = 'admin'` here?

Reply:
Wasn't there a validation like this` validate that the user is a student and return 404 if it isn't.`

Reply:
I made a mistake while writing the text - it should be an admin since only admin can update his own profile.